src/fiatconnect/saga.ts

```diff
@@ -67,6 +76,23 @@ export function* watchFetchFiatConnectQuotes() {
   yield takeLeading(fetchFiatConnectQuotes.type, handleFetchFiatConnectQuotes)
 }
 
+export function* handleFetchFiatConnectProviders() {
+  const account: string = yield select(currentAccountSelector)
+  try {
+    if (!account) {
+      throw new Error('Cannot fetch fiatconnect providers without an account')
+    }
+    const providers: FiatConnectProviderInfo[] = yield call(getFiatConnectProviders, account)
+    yield put(fetchFiatConnectProvidersCompleted({ providers }))
+  } catch (error) {
+    Logger.error(TAG, 'Error in *handleFetchFiatConnectProviders ', error)
+  }
+}
+
+export function* watchFetchFiatConnectProviders() {
+  yield takeLeading(fetchFiatConnectProviders.type, handleFetchFiatConnectProviders)
```

Comment:
just for my understanding, this ensures only one call can happen even if someone navigates quickly from Add / withdraw to another screen and back to Add / withdraw right?

Reply:
takeLeading stops listening for more events of that type while the current function is executing, so yes.